### PR TITLE
docs: rewrite new package manager questionnaire

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes

- More descriptive titles
- More checkboxes to select the correct answer
- Styling fixes
- Drop `---` separator between sections

## Context

The current form is hard to read and use. I'm trying to improve it with this PR. I'm not closing any issue with this.

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: docs/development/new-package-manager-template.md [hunk 1]
```diff
@@ -1,80 +1,72 @@
...
-- [ ] Yes
-- [ ] No
+### Do many users need to extend the [`fileMatch`](../usage/configuration-options.md#filematch) pattern for custom file names?
 
-### Is the fileMatch pattern likely to get many "false hits" for files that have nothing to do with package management?
+- [ ] Yes.
+- [ ] No.
 
----
+### Is the `fileMatch` pattern going to get many "false hits" for files that have nothing to do with package management?
```
Comment (HonkingGoose) [inline docs/development/new-package-manager-template.md:31]:
```suggestion
### Is the [`fileMatch`](../usage/configuration-options.md#filematch) pattern going to get many "false hits" for files that have nothing to do with package management?
```

File: docs/development/new-package-manager-template.md [hunk 2]
```diff
@@ -1,80 +1,72 @@
...
 
-### Is it likely that many users would need to extend this pattern for custom file names?
+### Which [`fileMatch`](../usage/configuration-options.md#filematch) pattern(s) should be used?
 
-- [ ] Yes
-- [ ] No
+### Do many users need to extend the [`fileMatch`](../usage/configuration-options.md#filematch) pattern for custom file names?
 
-### Is the fileMatch pattern likely to get many "false hits" for files that have nothing to do with package management?
+- [ ] Yes.
```
Comment (HonkingGoose) [inline docs/development/new-package-manager-template.md:28]:
```suggestion
- [ ] Yes, provide details.
```

File: docs/development/new-package-manager-template.md [hunk 3]
```diff
@@ -1,106 +1,155 @@
...
-### Name of package manager
+### What's the name of the package manager?
 
-### What language does this support?
+### What language(s) does this package manager support?
 
 ### How popular is this package manager?
 
-### Does this language have other (competing?) package managers?
+### Are there other package managers for this language?
```
Comment (rarkins) [inline docs/development/new-package-manager-template.md:15]: I think people understand _competing_ better than this. Our goal is to understand if this new manager is a substitute for others

Comment (rarkins) [reply docs/development/new-package-manager-template.md:15]: Maybe we can ask if this manager is used as an alternative or competes with any existing supported managers which Renovate has

**Review by rarkins — APPROVED** [review]

**Pull request merged**